community/modules/scripts/htcondor-install/files/autoscaler.py

```diff
@@ -226,29 +232,21 @@ def scale(self):
             print("Information about the compute instance template")
             pprint(instanceTemplateInfo)
 
-        # Calculate number instances to satisfy current job queue CPU requests
-        if queue > 0:
-            self.size = int(math.ceil(float(queue) / float(self.cores_per_node)))
-            if self.debug > 0:
-                print(
-                    "Calculating size of MIG: ⌈"
-                    + str(queue)
-                    + "/"
-                    + str(self.cores_per_node)
-                    + "⌉ = "
-                    + str(self.size)
-                )
-        else:
-            self.size = 0
-
-        # If compute instance limit is specified, can not start more instances then specified in the limit
-        if self.compute_instance_limit > 0 and self.size > self.compute_instance_limit:
-            self.size = self.compute_instance_limit
-            print(
-                "MIG target size will be limited by " + str(self.compute_instance_limit)
-            )
-
-        print("New MIG target size: " + str(self.size))
+        # Calculate the minimum number of instances that, for fully packed
+        # execute points, could satisfy current job queue
+        min_hosts_for_queue = int(math.ceil(float(queue) / float(self.cores_per_node)))
```

Comment:
Rest of code isn't Python2-compatible. In Python3 `/`-op does "true division", also `math.ceil` returns int.
```python
min_hosts_for_queue = math.ceil(queue / self.cores_per_node)
```

Reply:
Where I am using existing code, I've mostly left it alone, but I agree that this merits updating. I have simultaneously updated the shebang to python3.

Reply:
This was updated in a subsequent commit. I will resolve the conversation.